src/hooks/tests/use-update-demo-site.test.tsx

```diff
@@ -128,6 +128,106 @@ describe( 'useUpdateDemoSite', () => {
 		expect( result.current.isDemoSiteUpdating( mockLocalSite.id ) ).toBe( false );
 	} );
 
+	it( 'should allow updating two sites independently with different completion times', async () => {
+		const mockSnapshot2: Snapshot = {
+			atomicSiteId: 67890,
+			localSiteId: '09876',
+			url: '',
+			date: 0,
+		};
+		const mockLocalSite2: SiteDetails = {
+			name: 'Test Site 2',
+			running: false,
+			phpVersion: '8.0',
+			id: '09876',
+			path: '/path/to/site2',
+		};
+
+		// Mock different response times for each site
+		clientReqPost.mockImplementation( ( args ) => {
+			if ( args.formData[ 0 ][ 1 ] === mockSnapshot.atomicSiteId ) {
+				return new Promise( ( resolve ) =>
+					setTimeout( () => resolve( { data: 'success' } ), 1000 )
+				);
+			} else {
+				return new Promise( ( resolve ) =>
+					setTimeout( () => resolve( { data: 'success' } ), 2000 )
+				);
+			}
+		} );
+
+		const { result } = renderHook( () => useUpdateDemoSite(), { wrapper } );
+
+		// Start updating both sites
+		await act( async () => {
+			result.current.updateDemoSite( mockSnapshot, mockLocalSite );
+			result.current.updateDemoSite( mockSnapshot2, mockLocalSite2 );
+		} );
+
+		// Initially, both sites should be marked as updating
+		expect( result.current.isDemoSiteUpdating( mockLocalSite.id ) ).toBe( true );
+		expect( result.current.isDemoSiteUpdating( mockLocalSite2.id ) ).toBe( true );
+
+		// Wait for the first site to complete
+		await act( async () => {
+			jest.advanceTimersByTime( 1000 );
+			await Promise.resolve();
+		} );
+
+		// After 1000ms, the first site should be done, but the second should still be updating
+		expect( result.current.isDemoSiteUpdating( mockLocalSite.id ) ).toBe( false );
+		expect( result.current.isDemoSiteUpdating( mockLocalSite2.id ) ).toBe( true );
+
+		// Wait for the second site to complete
+		await act( async () => {
+			jest.advanceTimersByTime( 1000 );
+			await Promise.resolve();
+		} );
```

Comment:
This could be simplified by using `advanceTimersByTimeAsync`.

```suggestion
		await act( async () => jest.advanceTimersByTimeAsync( 1000 ) );
```

Reply:
I tried to make these changes but for some reason the test is failing with this setup 🤔  This is an example of failure:

```
  ● useUpdateDemoSite › should allow updating two sites independently with different completion times

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

      175 |
      176 |             // After 1000ms, the first site should be done, but the second should still be updating
    > 177 |             expect( result.current.isDemoSiteUpdating( mockLocalSite.id ) ).toBe( false );
          |                                                                             ^
      178 |             expect( result.current.isDemoSiteUpdating( mockLocalSite2.id ) ).toBe( true );
      179 |
      180 |             // Wait for the second site to complete

      at src/hooks/tests/use-update-demo-site.test.tsx:177:67
      at fulfilled (src/hooks/tests/use-update-demo-site.test.tsx:5:58)
```
      
I am wondering if there is a difference between how `jest.advanceTimersbyTime` works compared to `advanceTimersByTimeAsync` in executing scheduled callbacks.
     
I have not checked yet in detail but I saw this issue that might be related: https://github.com/jestjs/jest/issues/14928
     
@fluiddot do you happen to know why the test might fail in that case?

Reply:
I tried debugging this a bit more with `Alland when I log out states, I can see the following:

```
console.log
    Initial state: true true

      at src/hooks/tests/use-update-demo-site.test.tsx:167:11

  console.log
    After 1000ms: false true

      at src/hooks/tests/use-update-demo-site.test.tsx:177:11

  console.log
    In waitFor: false true

      at timeout (src/hooks/tests/use-update-demo-site.test.tsx:185:13)

  console.log
    After waitFor: false true

```

so it seems that everything is working correctly with `advanceTimersByTimeAsync` but there might not be something that is not resolved yet so the second site does not set to `false` on time.

Reply:
> @fluiddot do you happen to know why the test might fail in that case?

Ah, that's interesting. In my case, all tests are passing after applying the change 🤔. In any case, if the test results differ between environments, this might be a signal that it could fail in the future. That said, feel free to omit these changes.